### ISSUE TITLE
Backup/Restore fails on MySQL 8.0.x

### ISSUE THREAD

**Issue by derekperkins** [vitessio/vitess#4307]
Running a backup with MySQL 8 exits without failure, but the file manifest is dramatically different / shorter than the equivalent 5.7 backup.

The culprit appears to be the data dictionary changes that no longer require `.frm` files.
> Look for at least one .frm file
https://github.com/vitessio/vitess/blob/master/go/vt/mysqlctl/backup.go#L167-L178

We'll need to either add another check in the same loop to determine if it is a dbDir, or otherwise make a determination. Are there any other data dictionary changes we should be aware of? @sjmudd 

### MySQL 8.0 Backup Manifest

```json
{
FileEntries: [
{
Base: "InnoDBData",
Name: "ib_buffer_pool",
Hash: "8e6249d5"
},
{
Base: "InnoDBData",
Name: "ibdata1",
Hash: "c4317ee2"
},
{
Base: "InnoDBLog",
Name: "ib_logfile0",
Hash: "64d574d1"
},
{
Base: "InnoDBLog",
Name: "ib_logfile1",
Hash: "033b25d3"
}
],
Position: "MySQL56/bd077f94-d6fb-11e8-9293-3afc8af88f99:1-6",
TransformHook: "",
SkipCompress: false
}
```

### MySQL 5.7 Backup Manifest
https://gist.github.com/derekperkins/9f0683f07f97f65a4410b64a4689496c

### MySQL 8.0 Failed Restore Logs
https://gist.github.com/derekperkins/877f23c7e8bedaab009c332275b7402e

**Comment by derekperkins** [comment]
After deploying with the small change in #4308, the backup picked up a few more files, but still didn't restore properly.
```
{
FileEntries: [
{
Base: "InnoDBData",
Name: "ib_buffer_pool",
Hash: "b5978454"
},
{
Base: "InnoDBData",
Name: "ibdata1",
Hash: "4d29e70c"
},
{
Base: "InnoDBLog",
Name: "ib_logfile0",
Hash: "6baa2f8a"
},
{
Base: "InnoDBLog",
Name: "ib_logfile1",
Hash: "033b25d3"
},
{
Base: "Data",
Name: "_vt/local_metadata.ibd",
Hash: "603bad72"
},
{
Base: "Data",
Name: "_vt/reparent_journal.ibd",
Hash: "33d6c46b"
},
{
Base: "Data",
Name: "_vt/shard_metadata.ibd",
Hash: "abe2b295"
},
{
Base: "Data",
Name: "sys/sys_config.ibd",
Hash: "f06d02a3"
}
],
Position: "MySQL56/24630385-d864-11e8-92b3-0aea23f94457:1-6",
TransformHook: "",
SkipCompress: false
}
```

**Comment by derekperkins** [comment]
After running a restore, this is the error that I'm getting:

```
$ Failed to find valid data directory.
$ Data Dictionary initialization failed.
$ Aborting
$ /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.13)  MySQL Community Server - GPL.
```
Here's a file level comparison, so I think if we just backup the `mysql` directory and `mysql.ibd`, we should be good.
```
Before Backup        | After Restore
-----------------------------------------
#innodb_temp         | 
_vt                  | _vt
auto.cnf             | 
ca-key.pem           | 
ca.pem               | 
client-cert.pem      | 
client-key.pem       | 
mysql                | 
mysql.ibd            | 
performance_schema   | 
private_key.pem      | 
public_key.pem       | 
server-cert.pem      | 
server-key.pem       | 
sys                  | sys
undo_001             | 
undo_002             | 
vt_unsharded80       | vt_unsharded80
```

**Comment by derekperkins** [comment]
Solved in the latest PR push to #4308
